client/my-sites/upgrades/cart/cart-trial-ad.jsx

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { cartItems } from 'lib/cart-values';
+import { addCurrentPlanToCartAndRedirect, getCurrentPlan, getDaysSinceTrialStarted } from 'lib/plans';
+import i18n from 'lib/mixins/i18n';
+
+const CartTrialAd = ( { cart, sitePlans, selectedSite } ) => {
+	const isDataLoading = ! sitePlans.hasLoadedFromServer || ! cart.hasLoadedFromServer,
+		currentPlan = getCurrentPlan( sitePlans.data );
+
+	if ( isDataLoading ||
+		! currentPlan.freeTrial ||
+		cartItems.getDomainRegistrations( cart ).length !== 1 ) {
+		return <noscript />;
```

Comment:
Stateless components can't return `null`, and React uses `<noscript />` when stateful components that return `null`. I updated the first commit with a comment about this behavior and a link to the relevant GitHub issue.